data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/debug/GetDeposits.java

```diff
@@ -0,0 +1,130 @@
+
+  private static final SerializableTypeDefinition<List<Deposit>> DEPOSITS_RESPONSE_TYPE =
+      SerializableTypeDefinition.<List<Deposit>>object()
+          .name("GetDepositsResponse")
+          .withField(
+              "data", listOf(Deposit.SSZ_SCHEMA.getJsonTypeDefinition()), Function.identity())
+          .build();
+
+  private final ChainDataProvider chainDataProvider;
+  private final DepositProvider depositProvider;
+  private final Spec spec;
+
+  public GetDeposits(
+      final DataProvider provider, final DepositProvider depositProvider, final Spec spec) {
+    this(provider.getChainDataProvider(), depositProvider, spec);
+  }
+
+  public GetDeposits(
+      final ChainDataProvider chainDataProvider,
+      final DepositProvider depositProvider,
+      final Spec spec) {
+    super(
+        EndpointMetadata.get(ROUTE)
+            .operationId("getDeposits")
+            .summary("Deposits not yet included on the chain")
+            .description(
+                "Range of deposits that could be used in new block if we are assigned to be a proposer.")
+            .tags(TAG_DEBUG)
+            .response(SC_OK, "Request successful", DEPOSITS_RESPONSE_TYPE)
+            .build());
+    this.chainDataProvider = chainDataProvider;
+    this.depositProvider = depositProvider;
+    this.spec = spec;
+  }
+
+  @OpenApi(
+      path = ROUTE,
+      method = HttpMethod.GET,
+      summary = "Deposits not yet included on the chain",
+      description =
+          "Range of deposits that could be used in new block if we are assigned to be a proposer.",
+      tags = {TAG_DEBUG},
+      responses = {
+        @OpenApiResponse(
+            status = RES_OK,
+            content = @OpenApiContent(from = GetDepositsResponse.class)),
+        @OpenApiResponse(status = RES_NOT_FOUND),
+        @OpenApiResponse(status = RES_INTERNAL_ERROR)
+      })
+  @Override
+  public void handle(@NotNull final Context ctx) throws Exception {
+    adapt(ctx);
+  }
+
+  @Override
+  public void handleRequest(final RestApiRequest request) throws JsonProcessingException {
+    request.respondAsync(
+        chainDataProvider
+            .getBeaconState(DEFAULT_STATE_ID)
+            .thenApplyChecked(
+                maybeState -> {
+                  if (maybeState.isEmpty()) {
+                    return AsyncApiResponse.respondNotFound();
+                  } else {
+                    List<Deposit> allDeposits =
+                        depositProvider.getAllPendingDeposits(maybeState.get().getData());
+                    return AsyncApiResponse.respondOk(allDeposits);
+                  }
+                })
+            .exceptionally(
+                ex -> {
+                  LOG.error("Failed to retrieve deposits", ex);
+                  return AsyncApiResponse.respondWithError(
+                      SC_INTERNAL_SERVER_ERROR,
+                      "An internal error occurred, check the server logs for more details.");
+                }));
```

Comment:
We should never be throwing a SC_INTERNAL_SERVER_ERROR in a handler, that literally is declaring we have a bug.
let the default handler deal with unexpected errors... is there a case this happens? because that would indicate we've got a problem...
also, can log.debug here if we really had to, but I would be inclined to simply delete this exceptionally. If we were to do something here we'd not be logging an error.

Reply:
@rolfyone I could catch it for example when deposits are not yet loaded, if sending query in the very beginning

Reply:
this will literally become an internal server error, an internal server error is what happens when an exception is uncaught, and should only ever be put in by the default exception handler.
what specific exception do we need to handlehere if deposits arent loaded?
`ChainDataUnavailableException`, `ServiceUnavailableException` and a number of others are handled generically, so I'm unclear as to why we'd be catching here unless we're looking for something, and if we're looking for something, the result is *never* going to be a sc_internal_server_error.

For specific errors, it may be worth putting a 503 (service unavailable) during startup scenario, but i don't want a generic exceptionally that catches all errors and throws it to internal error.

More likely would be that the function being called raises something appropriate - its a new function for our use. so if we find that the service isn't ready, throw ServiceUnavailableException in deposit provider and no specific handling would be required here in the handler (other than documenting a 503 may happen)